### ISSUE TITLE
Lab 03 Too many service times generated in CodeCrunch

### ISSUE THREAD

**Issue by syafiq9326** [nus-cs2030/2223-s2#100]
# Summary

Code works fine from my side, throws in CodeCrunch to get the above error.

## Description:

Hi, may I know if there is some sort of limit or boundaries when using the get from DefaultServiceTime as I think that the above error is due to me accidentally overcalling it? 

## Screenshots

![lab3_github](https://user-images.githubusercontent.com/50677984/218176051-77817248-fbd7-4e73-96af-0b792528d5e9.png)

**Comment by maze508** [comment]
If you refer to **Some Design Tips...**, the last  segment of code crunch, it says :

- In your design, only invoke getServiceTime() in one location. Invoking in multiple locations may result in wrong service time.

**Comment by syafiq9326** [comment]
Ahh okay, I will try to switch it up, thanks!

**Comment by leezhanpeng** [comment]
Yup it is as it says. Your programme called for the customer's service duration more times than necessary. I made the same error from not properly capping the queue size, ended up serving more customers than needed. May or may not be relevant to your case, good luck!

**Comment by poon-zx** [comment]
I also encountered a similar error but I only called getServiceTime() once in the Arrival class. But I assigned it to a new variable which I passed into my Serve and Wait classes, is that not allowed?

**Comment by leezhanpeng** [comment]
> I also encountered a similar error but I only called getServiceTime() once in the Arrival class. But I assigned it to a new variable which I passed into my Serve and Wait classes, is that not allowed?

It could be that you called getServiceTime() for customers that are supposed to leave without getting served.

**Comment by fy17ohhh** [comment]
hi, I have encountered the same issue. Does this mean that we can only use getServiceTime() exactly once in all files?

**Comment by maze508** [comment]
Not necessarily but you need to make sure you call it only once for each Customer I believe.

**Comment by jhsoh** [comment]
> hi, I have encountered the same issue. Does this mean that we can only use getServiceTime() exactly once in all files?

You can save it in a variable if you need to use it multiple times in a class.

**Comment by shuyaoao** [comment]
I also faced the same issue :(( I only used serviceTime.get() in my Arrival class getNextEvent method twice under two mutually exclusive conditions so it should only call once for every Arrival event initialised, not sure why it doesn't work...

**Comment by EvanCh08** [comment]
Hello, for this case, you only need to invoke getServiceTime() once, my suggestion is to store it in a variable so that it will not be changed when you call that variable inside your customer class.

**Comment by wadeyman** [comment]
perhaps you can call the getservicetimes once and store it else where in your variable which you retrieve from in the future

**Comment by VanyaPriscillia** [comment]
Yes, error occurs in the first test case if you call serviceTime.get() more than necessary. I think you should recheck if there's anything you haven't updated.

**Comment by syafiq9326** [comment]
How I resolved the error was to only called my getServiceTime() under serveEvent. Ultimately, from what I have seen, the number of ocurrences get ServiceTime() is called should correspond to the number of serve events in codecrunch test cases (which would explain service time errors occurring only at certain test cases as you might have generated extra serve events). 

I think if you are able to clear the first few test cases, the problem should now tie back to your serve events and no longer an issue with the getServiceTime().

To help with finding my errors, I did a system print under my getServiceTime() method which should be under your customer class. The number of times it should be printed out must correspond to the quantity of your serve events.
